src/feature/dashboard/presentation/dashboard.screen.tsx

```diff
@@ -1,10 +1,280 @@
+import { format } from 'date-fns';
+// import { UserContext } from '../../core/context/UserContext';
+import { useLocalStorage } from 'core/localStorage/localStorage.hook';
+import { UserProfile } from 'core/types';
 
 export const DashboardScreen = (): JSX.Element => {
+  const [isBorrowModalOpen, setIsBorrowModalOpen] = useState(false);
+  const [reservedBook, setReservedBook] = useState<IBook | null>(null);
+  const [reservedAuthor, setReservedAuthor] = useState<IAuthor | null>(null);
+  const [selectedDate, setSelectedDate] = useState<Date>(new Date());
+  const [isSnackbarOpen, setIsSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+  const [searchData, setSearchData] = useState('');
+  const { books } = useBooksData();
+  const { authors } = useAuthorsData();
+  const { canBookBeReserved, isFetching, reservationDueDate } =
+    useCanBookBeReserved(reservedBook, selectedDate);
+  const [localUser] = useLocalStorage<UserProfile | null>('user', null);
+
+  const reserveBook = async () => {
+
+    if (!reservedBook || !localUser) {
+      return;
+    }
+
+    // TODO: check reservation
+
+    let random = Math.round(Math.random()); // TODO: this is only mockup. Remove it later.
+
+    if (random) {
+      setSnackbarMessage('Book reserved successfully!');
+      setIsSnackbarOpen(true);
+      setIsBorrowModalOpen(false);
+      setReservedBook(null);
+    } else {
+      setSnackbarMessage('Something went wrong. Try again later.');
+      setIsSnackbarOpen(true);
+      setIsBorrowModalOpen(false);
+      setReservedBook(null);
+    }
+  };
+
   return (
-    <Paperbase>
-      <div>Dashboard</div>
-    </Paperbase>
+    <>
+      <Paperbase>
+        <Paper sx={{ maxWidth: 936, margin: 'auto', overflow: 'hidden' }}>
+          <AppBar
+            position="static"
+            color="default"
+            elevation={0}
+            sx={{ borderBottom: '1px solid rgba(0, 0, 0, 0.12)' }}
+          >
+            <Toolbar>
+              <Grid container spacing={2} alignItems="center">
+                <Grid item>
+                  <SearchIcon color="inherit" sx={{ display: 'block' }} />
+                </Grid>
+                <Grid item xs>
+                  <TextField
+                    fullWidth
+                    placeholder="Search through books by title, author or ISBN"
+                    InputProps={{
+                      disableUnderline: true,
+                      sx: { fontSize: 'default' },
+                    }}
+                    variant="standard"
+                    onChange={(e) => {
+                      setSearchData(e.target.value);
+                    }}
+                    value={searchData}
+                  />
+                </Grid>
+              </Grid>
+            </Toolbar>
+          </AppBar>
+          <Grid container spacing={4} sx={{ p: 6 }}>
+            <TableContainer>
+              <Table>
+                <TableHead>
```

Comment:
The file gets pretty big, maybe you could move whole `TableHead` component to another file, for example `src/feature/dashboard/presentation/booksTableHeader.component.tsx`?